File: Application/DTOs/ProjectDto.cs
using Application.Common;

namespace Application.DTOs;

public class ProjectDto : BaseDto<int>
{
    public string Name { get; set; }
    public string Description { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public int StatusId { get; set; }
    public ICollection<TaskDto> Tasks { get; set; }
    public ICollection<EmployeeDto> Employees { get; set; }
    public ICollection<ManagerDto> Managers { get; set; }
}

File: UI/src/app/services/dtos/project.dto.ts
import {TaskDto} from "./task.dto";
import {EmployeeDto} from "./employee.dto";
import {ManagerDto} from "./manager.dto";

export interface ProjectDto {
  id: number;
  statusId: number;
  name: string;
  description: string;
  startDate: Date;
  endDate: Date;
  tasks: TaskDto[];
  employees: EmployeeDto[];
  managers: ManagerDto[];
}
